raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -116,7 +117,7 @@ public void initialize(OffsetAndEpoch logEndOffsetAndEpoch) throws IOException,
             if (election == null) {
                 election = ElectionState.withUnknownLeader(0, voters);
             }
-        } catch (final IOException e) {
+        } catch (final Exception e) {
```

Comment:
Why should we change `IOException` into `Exception` here?

Reply:
Previously, the IOException would have been thrown here, so you need to catch it here. After the KIP handles the IOException, the underlying IOException will not be thrown up, but will become other exceptions, such as UnCheckeDioException, so you need to handle it this way.

Reply:
Yes, but I think `Exception` is covering too many "unexpected" exceptions. You can see the catch block, we are handling IOException case, not other exceptions. I think we can directly catch `UncheckedIOException` here.

Reply:
Yes, I seem to have